// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// If you change or add any fields in this file, update proto_visitors.h and
// potentially proto_enum_conversions.{h, cc}.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.sync.protocol";

option optimize_for = LITE_RUNTIME;

package sync_pb;

// Valuables coming from Google Wallet.
message AutofillValuableSpecifics {
  // The valuable id.
  optional string id = 1;

  message LoyaltyCard {
    // The merchant name e.g. "Deutsche Bahn".
    optional string merchant_name = 1;

    // The program name e.g. "BahnBonus".
    optional string program_name = 2;

    // The logo URL.
    optional string program_logo = 3;

    // The loyalty card text code.
    optional string loyalty_card_number = 4;

    // Merchant domains associated to the loyalty card.
    repeated string merchant_domains = 5;
  }

  oneof valuable_data {
    LoyaltyCard loyalty_card = 2;
    // add other valuable types here.
  }
}
